lib/Pool.js

```diff
@@ -254,13 +260,14 @@ class Pool {
       for (let i = 0; i < count; i += 1) {
         let promise = new Promise((resolve, reject) => {
           odbc.connect(connectionConfig, (error, nativeConnection) => {
-            let connection = undefined;
+            this.connectingCount--;
```

Comment:
I moved the decrement of `connectingCount` before the error check because I think if there was an error making the connection we also want to decrement `connectingCount`.